### PR TITLE
ICDS data validations

### PR DESCRIPTION
**Pull request by emord**
Builds off the validation work. Thought it would be good to have a real use case before merging the other.

Going to go through these a second time tomorrow morning to be sure I'm making the correct assumptions

### CODE REVIEW

File: corehq/apps/userreports/README.rst [hunk 1]
```diff
@@ -1411,6 +1411,40 @@ used as a guide). There are also additional examples in the `practical examples`
        ]
    }
 
+Data Cleaning and Validation
```
Comment (emord) [inline corehq/apps/userreports/README.rst]: Not sure why this is showing up in the diff as its from the other PR.

**Review by calellowitz — APPROVED** [review]
This looks super cool. I did not check the actual UCR validations super closely (can go back if you prefer), but the framework seems great

**Review by snopoke — APPROVED** [review]
This is great.

File: custom/icds_reports/ucr/tests/test_beneficiary_form.py [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
-
 from mock import patch
+
 from custom.icds_reports.ucr.tests.test_base_form_ucr import BaseFormsTest
 
 
+@patch('custom.icds_reports.ucr.expressions._get_user_location_id',
+       lambda user_id: 'qwe56poiuytr4xcvbnmkjfghwerffdaa')
+@patch('corehq.apps.locations.ucr_expressions._get_location_type_name',
+       lambda loc_id, context: 'supervisor')
```
Comment (emord) [inline custom/icds_reports/ucr/tests/test_beneficiary_form.py]: I attempted to pull these patches out to some common code, but I was getting into an issue where the tests were no longer being picked up by the auto discover. Decided to punt for now

**Comment by emord** [comment]
Removed the validations from tableau data source because its being removed soon https://github.com/dimagi/commcare-hq/pull/23606 and i was having issues figuring out where to patch the tests

**Pull request merged**